lib/Dialect/Pulse/Transforms/Scheduling.cpp

```diff
@@ -178,30 +175,38 @@ void QuantumCircuitPulseSchedulingPass::scheduleAlap(
                                            totalDurationOfQuantumCircuit);
 }
 
-int QuantumCircuitPulseSchedulingPass::getNextAvailableTimeOfPorts(
-    mlir::ArrayAttr ports) {
-  int nextAvailableTimeOfAllPorts = 0;
-  for (auto attr : ports) {
-    const std::string portName = attr.dyn_cast<StringAttr>().getValue().str();
-    if (portName.empty())
-      continue;
-    if (portNameToNextAvailabilityMap.find(portName) !=
-        portNameToNextAvailabilityMap.end()) {
-      if (portNameToNextAvailabilityMap[portName] < nextAvailableTimeOfAllPorts)
-        nextAvailableTimeOfAllPorts = portNameToNextAvailabilityMap[portName];
+int64_t QuantumCircuitPulseSchedulingPass::getNextAvailableTimeOfMixFrames(
+    std::unordered_set<uint> &mixFramesBlockIds) {
+  int64_t nextAvailableTimeOfAllMixFrames = 0;
+  for (auto id : mixFramesBlockIds) {
+    if (mixFrameToNextAvailabilityMap.find(id) !=
+        mixFrameToNextAvailabilityMap.end()) {
+      if (mixFrameToNextAvailabilityMap[id] < nextAvailableTimeOfAllMixFrames)
```

Comment:
It seems like this condition can be combined into a single boolean and `if`? (can use an intermediary value)